### DISCUSSION TITLE
Custom Vega-Lite spec?

### DISCUSSION THREAD

**Discussion by jwhitaker-gridcog** [malloydata/malloy#1642] [Q&A] [answered]
Hey guys,
Is there any way I can control plotting by writing a vega lite spec instead of relying on the '# bar' etc annotations?

I can see a note
> The vega renderer allows much more customization of rendering than the default visualization options provided in the Extension, using the [Vega-Lite](https://vega.github.io/vega-lite/) library. For examples of using these in Malloy, check out the flights_custom_vis model and styles files in the FAA [Sample Models](https://docs.malloydata.dev/documentation/samples) download.

at the bottom of the Rendering doco, but that link looks broken and I can't find that file in github org search. :(

This is when using the VS code extension.
Cheers!

**Comment by skokenes** [comment, marked as answer]
Hey! Unfortunately, those docs are outdated. We don't currently support providing your own Vega specs to Malloy :(. I've updated those docs now, sorry for the confusion.

The good news: we are working on our next-gen data visualization renderer which will allow highly customized viz to be created easily with Malloy tags. See these blog posts to learn more about where we are headed:
- [Dataviz is Hierarchical](https://docs.malloydata.dev/blog/2024-02-29-hierarchical-viz)
- [Malloy Renders Nested Data](https://docs.malloydata.dev/blog/2024-02-08-tables)

We also are considering creating plugins that users can use to bring their own renderers like eCharts and custom vega specs. To follow along with the viz journey and give us feedback, consider joining us on the visualization channel in the [Malloy Community Slack](https://join.slack.com/t/malloy-community/shared_invite/zt-1kgfwgi5g-CrsdaRqs81QY67QW0~t_uw).

**Reply by jwhitaker-gridcog** [reply]
thanks! random user story: I would leap on a vega / vega-lite plugin if it (and the mechanism to allow it) became available.

The things you are playing with around visualization are super-interesting but I can't help but wonder how much complexity you're walking into there - dedicated viz projects like ggplot, vega, and vega-lite are incredibly complex on their own, and I don't know how core visualization is to your mission compared with the Malloy language itself.

**Reply by jwhitaker-gridcog** [reply]
also, unrelated, but the slack link here + in readme is expired :(

**Reply by skokenes** [reply]
You're not the only person that has asked to bring your own vega spec, so your random user story definitely counts and adds to the priority

Sorry about the Slack link, here is an updated one https://join.slack.com/t/malloy-community/shared_invite/zt-2crkkvo36-JNG6FM8cghRTyYmeS8hFBQ
